### DISCUSSION TITLE
[Help] `{{path}}`, but without file name

### DISCUSSION THREAD

**Discussion by mawkler** [zk-org/zk#342] [Help]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

I would like to list notes on the format:

```
{{path-without-filename}}/{{title}}
```

For example, given the following note with the path (from notebook root) `foo/bar/test/`:

```markdown
---
title: My Test Note
---
```

I want the output

```
foo/bar/test/My Test Note
```

Is there any way to achieve this?

**Comment by mawkler** [comment]
~~I found it! [`{{filename-stem}}`](https://github.com/mickael-menu/zk/blob/0b4db9ade640d99257d8c801e1b924ecb7013e05/docs/template-format.md#template-context-when-formatting-a-note)~~

Edit: never mind. `{{filename-stem}}` still includes the file name, it just excludes the file extension.

**Comment by mickael-menu** [comment]
Nothing out-of-the-box, but you can rely on shell commands, such as `dirname`:

```
zk list --format "{{sh (concat 'dirname ' path)}} {{title}}"
```

**Reply by mawkler** [reply]
@mickael-menu Thank you! However, with your solution, notes in the root directory get prepended with `.`. Is it possible exclude the path if it is `.`? I tried this:

```sh
zk list --format "
{{#if (not (eq (sh (concat 'dirname ' path)) '.'))}}
  {{sh (concat 'dirname ' path)}}
{{/if}}
"
```

But it doesn't seem like `zk`'s templating system supports `eq`. Do you know any way to achieve this?
